packages/nimble-components/src/rich-text/viewer/tests/rich-text-viewer.spec.ts

```diff
@@ -81,4 +125,268 @@ describe('RichTextViewer', () => {
+            ]);
+            expect(pageObject.getRenderedMarkdownLastChildContents()).toBe(
+                'user:1'
+            );
+        });
+
+        it('removing mapping element renders the mention node with user ID', async () => {
+            element.markdown = '<user:1>';
+            await appendUserMentionConfiguration(
+                element,
+                ['user:1'],
+                ['username1']
+            );
+
+            const renderedUserMention = element.firstElementChild as RichTextMentionUsers;
+            const renderedMappingUser = renderedUserMention.firstElementChild as MappingUser;
+            renderedUserMention.removeChild(renderedMappingUser);
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getRenderedMarkdownTagNames()).toEqual([
+                'P',
+                `${richTextMentionUsersViewTag}`.toUpperCase()
+            ]);
+            expect(
+                pageObject.getRenderedMarkdownLastChildAttribute(
+                    'mention-label'
+                )
+            ).toEqual('1');
+        });
+
+        it('updating to the `pattern` in mention configuration converts the mention to absolute link', async () => {
+            element.markdown = '<user:1>';
+            await appendUserMentionConfiguration(
+                element,
+                ['user:1'],
+                ['username1']
+            );
+
+            (element.firstElementChild as RichTextMentionUsers).pattern = 'invalid';
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getRenderedMarkdownTagNames()).toEqual([
+                'P',
+                `${anchorTag}`.toUpperCase()
+            ]);
+            expect(pageObject.getRenderedMarkdownLastChildContents()).toBe(
+                'user:1'
+            );
+        });
+
+        it('updating `display-name` in mapping mention should update the `mention-label` in view', async () => {
+            element.markdown = '<user:1>';
+            await appendUserMentionConfiguration(
+                element,
+                ['user:1'],
+                ['username1']
+            );
+
+            const renderedUserMention = element.firstElementChild as RichTextMentionUsers;
+            const renderedMappingUser = renderedUserMention.firstElementChild as MappingUser;
+            renderedMappingUser.displayName = 'updated-name';
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getRenderedMarkdownTagNames()).toEqual([
+                'P',
+                `${richTextMentionUsersViewTag}`.toUpperCase()
+            ]);
+            expect(
+                pageObject.getRenderedMarkdownLastChildAttribute(
+                    'mention-label'
+                )
+            ).toEqual('updated-name');
+        });
+
+        it('updating valid `key` in mapping mention should update it to a mention view if it is a absolute link before', async () => {
```

Comment:
Could you add a test for the opposite of this case? i.e. change the key so that a mention which was previously unmatched becomes matched, converting it from a link to a mention view.

Reply:
Added a test case for the opposite case.